fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -343,8 +356,19 @@ private IndexingMultiTargetByRecords getIndexerMultiTargetByRecords() {
         return (IndexingMultiTargetByRecords)indexer;
     }
 
+    @Nonnull
+    private IndexingMutuallyByRecords getMutualIndexerByRecords() {
+        if (! (indexer instanceof IndexingMutuallyByRecords)) {
+            indexer = new IndexingMutuallyByRecords(common, indexingPolicy, indexingPolicy.mutualIndexingBoundaries);
+        }
+        return (IndexingMutuallyByRecords)indexer;
+    }
+
     @Nonnull
     private IndexingBase getIndexer() {
+        if (indexingPolicy.isMutual() && !fallbackToRecordsScan) {
```

Comment:
Why does this condition have `!fallbackToRecordsScan`? Is the idea that that parameter means that we want a vanilla `BY_RECORDS` build? It sees like it can chose to do a multi-index build in that case, but you want to preclude a mutual build?

Reply:
In this fallback path some of the indexes, but not all, became readable. 
This can be limited to one or few processes (because the current indexer happened to check states while another indexer was converting all the indexes to readable), or to all but one processes (if one of the indexes has a uniqueness violation). Chances are that there are no more missing ranges. 
Either way, processes in this path will fallback to a MULTI_TARGET indexing, where they'll acquire a lock and attempt to finish converting the non-readable indexes (if any). This aligns with the strategy of "making best effort" not to leave any index behind and can handle the case of another process crashing during the step of making all the indexes readable.

Reply:
I agree that the fallback logic should be simplified. Is that what is bothering you?